inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/permute/permute_kernel_tile_8x8_4x4.cpp

```diff
@@ -0,0 +1,330 @@
+            break;
+        default:
+            throw std::runtime_error("Unsupported combination\n");
+            break;
+    }
+    dispatchData.lws = GetBestLwsFromGws(params, dispatchData.gws, tile_size);
+    return dispatchData;
+}
+
+bool PermuteKernel_tile_8x8_4x4::Validate(const Params& p, const optional_params& o) const {
+    if (p.GetType() != KernelType::PERMUTE || o.GetType() != KernelType::PERMUTE) {
+        return false;
+    }
+
+    const permute_params& params = static_cast<const permute_params&>(p);
+
+    for (auto& fused_op : params.fused_ops) {
+        if (!IsFusedPrimitiveSupported(fused_op))
+            return false;
+    }
+
+    if (params.output.PitchesDifferFromLogicalDims() || params.inputs[0].PitchesDifferFromLogicalDims()) {
+        return false;
+    }
+
+    return true;
+}
+
+KernelsData PermuteKernel_tile_8x8_4x4::GetKernelsData(const Params& params, const optional_params& options) const {
+    if (!Validate(params, options)) {
+        return {};
+    }
+
+    KernelData kd = KernelData::Default<permute_params>(params);
+    permute_params& newParams = *static_cast<permute_params*>(kd.params.get());
+    size_t tile_size = DEFAULT_TILE_SIZE;
+    // supports 4x4 or 8x8 tiling
+    if (newParams.inputs[0].X().v < DEFAULT_TILE_SIZE || newParams.inputs[0].Feature().v < DEFAULT_TILE_SIZE)
+        tile_size = MIN_TILE_SIZE;
+
+    auto dispatchData = SetDefault(newParams, tile_size);
+    auto cldnn_jit = GetJitConstants(newParams, dispatchData, tile_size);
+    auto entry_point = GetEntryPoint(kernelName, newParams.layerID, options);
+    std::string jit = CreateJit(kernelName, cldnn_jit, entry_point);
+    auto& kernel = kd.kernels[0];
+    FillCLKernelData(kernel, dispatchData, params.engineInfo, kernelName, jit, entry_point, "", false, false, 1, GetFusedPrimitiveInputsCount(params));
+
+    return {kd};
+}
+
+KernelsPriority PermuteKernel_tile_8x8_4x4::GetKernelsPriority(const Params& params/*params*/, const optional_params& /*options*/) const {
+
+    std::function<bool(const std::vector<uint16_t>&)> is_rotating_except_batch = [](const std::vector<uint16_t>& order) {
+        // Target transform: Rotate feature dim to back to be taken as inner-most axis
+        // ex) 0(b), 4(f), 1(z), 2(y), 3(x)
+        if ((int32_t) order[1] != order.size() - 1) return false;
+        for (int32_t i = 3; i < (int32_t) order.size(); ++i) {
+            if ((int32_t)order[i] !=  (i - 1)) return false;
+        }
+        return true;
+    };
+
+    KernelData kd = KernelData::Default<permute_params>(params);
+    permute_params& newParams = *static_cast<permute_params*>(kd.params.get());
+
+    if (is_rotating_except_batch(newParams.order) && (newParams.inputs[0].Feature().v > MIN_TILE_SIZE) && (newParams.inputs[0].X().v > MIN_TILE_SIZE)) {
+        return FORCE_PRIORITY_1;
+    } else
+        return DONT_USE_IF_HAVE_SOMETHING_ELSE;
```

Comment:
Does the kernel support other permutation cases except this rotation? Maybe you should just reject other cases in Validate method?

Reply:
You are right. Will fix soon

Reply:
Fixed